next/pages/all-articles/[[...page]].tsx

```diff
@@ -0,0 +1,105 @@
+import { GetStaticPaths, GetStaticProps, InferGetStaticPropsType } from "next";
+import { useTranslation } from "next-i18next";
+import { useRef } from "react";
+
+import { ArticleListItem } from "@/components/article-list-item";
+import { HeadingPage } from "@/components/heading--page";
+import { LayoutProps } from "@/components/layout";
+import { Meta } from "@/components/meta";
+import { Pagination, PaginationProps } from "@/components/pagination";
+import { getLatestArticlesItems } from "@/lib/get-articles";
+import { getCommonPageProps } from "@/lib/get-common-page-props";
+import {
+  ArticleTeaser as ArticleTeaserType,
+  validateAndCleanupArticleTeaser,
+} from "@/lib/zod/article-teaser";
+
+interface AllArticlesPageProps extends LayoutProps {
+  articleTeasers: ArticleTeaserType[];
+  paginationProps: PaginationProps;
+}
+
+export default function AllArticlesPage({
+  articleTeasers = [],
+  paginationProps,
+}: InferGetStaticPropsType<typeof getStaticProps>) {
+  const { t } = useTranslation();
+  const focusRef = useRef<HTMLDivElement>(null);
+  return (
+    <>
+      <Meta title={t("all-articles")} metatags={[]} />
+      <div ref={focusRef} tabIndex={-1} />
+      <HeadingPage>{t("all-articles")}</HeadingPage>
+      <ul className="mt-4">
+        {articleTeasers?.map((article) => (
+          <li key={article.id}>
+            <ArticleListItem article={article} />
+          </li>
+        ))}
+      </ul>
+      <Pagination
+        focusRestoreRef={focusRef}
+        paginationProps={paginationProps}
+      />
+    </>
+  );
+}
+
+// eslint-disable-next-line @typescript-eslint/require-await
+export const getStaticPaths: GetStaticPaths = async () => {
+  return {
+    paths: [
+      {
+        params: { page: ["1"] },
+      },
+    ],
+    fallback: "blocking",
+  };
+};
+
+export const getStaticProps: GetStaticProps<AllArticlesPageProps> = async (
+  context
+) => {
+  // Get the page parameter:
+  const page = context.params.page;
+  const currentPage = parseInt(Array.isArray(page) ? page[0] : page || "1");
```

Comment:
this is a bit funky, but basically it you go to /all-articles without the "1" at the end this will turn the parameter into page 1